jdaviz/configs/imviz/plugins/catalogs/catalogs.py

```diff
@@ -81,25 +131,38 @@ def search(self):
             # finds all the sources in that region
             query_region_result = SDSS.query_region(skycoord_center, radius=zoom_radius,
                                                     data_release=17)
+            if query_region_result is None:
+                self.results_available = True
+                self.number_of_results = 0
+                self.app._catalog_source_table = None
+                return
+
             # TODO: Filter this table the same way as the actual displayed markers.
             # attach the table to the app for Python extraction
             self.app._catalog_source_table = query_region_result
-            self.results_available = True
+            skycoord_table = SkyCoord(query_region_result['ra'],
+                                      query_region_result['dec'],
+                                      unit='deg')
+
+        elif self.catalog_selected == 'From File...':
+            # all exceptions when going through the UI should have prevented setting this path
+            # but this exceptions might be raised here if setting from_file from the UI
+            table = self._cached_table_from_file.get(self.from_file, QTable.read(self.from_file))
+            self.app._catalog_source_table = table
+            skycoord_table = table['sky_centroid']
 
         else:
             self.results_available = False
             self.number_of_results = 0
+            self.app._catalog_source_table = None
             raise NotImplementedError(f"{self.catalog_selected} not a supported catalog")
 
-        # nothing happens in the case the query returned empty
-        if query_region_result is None:
+        self.results_available = True
+        if not len(skycoord_table):
             self.number_of_results = 0
+            self.app._catalog_source_table = None
             return
 
-        # TODO: Column names are catalog-dependent. Need to consider this when adding new catalogs.
```

Comment:
This TODO is also still relevant despite this patch.

Reply:
(same response as above... this is now covered by existing tickets/issues but I can revert if you prefer inline TODO comments)